wallet/rescan.go

```diff
@@ -285,6 +285,33 @@ func (w *Wallet) Rescan(ctx context.Context, n NetworkBackend, startHash *chainh
 	return nil
 }
 
+// RescanProgressFromHash starts a rescan of the wallet for all blocks on the main chain
+// beginning at startHash.  This function blocks until the rescan completes.
+func (w *Wallet) RescanProgressFromHash(ctx context.Context, n NetworkBackend, startHash *chainhash.Hash, p chan<- RescanProgress) {
```

Comment:
This function must have different semantics than simply starting a rescan on the main chain at the height of a block with hash startHash.  It must first verify that startHash has a saved header, and walk the header chain backwards until the header refers to a main chain block.  This will usually be startHash itself, but it's possible that it's not, and under that situation, the rescan must start at an earlier main chain height.